ccm_web/server/src/api/api.service.ts

```diff
@@ -21,4 +29,44 @@ export class APIService {
       }
     }
   }
+
+  async getCourseName (userLoginId: string, courseId: number): Promise<string | null> {
+    const requestor = await this.canvasService.createRequestorForUser(userLoginId)
+    let name = null
+    try {
+      const response = await requestor.get<Course>(`courses/${courseId}`)
+      const course = response.body
+      name = course.name
+    } catch (error) {
+      if (error instanceof HTTPError && error.response.statusCode !== HttpStatus.OK) {
+        logger.error(`Received unusual status code ${String(error.response.statusCode)}`)
+        logger.error(`Response body: ${JSON.stringify(error.response.body)}`)
+      } else {
+        logger.error(`An error occurred while making a request to Canvas: ${JSON.stringify(error)}`)
+      }
+      throw new Error(error)
+    }
+    return name
+  }
+
+  async putCourseName (userLoginId: string, courseId: number, newName: string): Promise<string | null> {
+    const requestor = await this.canvasService.createRequestorForUser(userLoginId)
+    let name = null
+    try {
+      const response = await requestor.requestUrl<Course>(
+        `courses/${courseId}`, 'PUT', { course: { name: newName } }
```

Comment:
One of things I heard from Stakeholders is when they change a course name they also change the course_code as well. So please include update of `course[course_code]` in the API call when updating. Course_code is same as course name. You can verify it with any Canvas lecture courses course_name and course_code is the same. I think backend can simply update the course_code when they change the course_name. 
I don't see a point to have course_code variable in the swagger API but I might be wrong. So I will leave it up to you.

Reply:
Is this request documented in the requirements somewhere? Did I miss it, or we adding it now?

Reply:
I did document this in the issue https://github.com/tl-its-umich-edu/canvas-course-manager-next/issues/26.

Reply:
You are correct, I will add that.

Reply:
Done. See https://github.com/tl-its-umich-edu/canvas-course-manager-next/pull/121/commits/f54ba63ca5b22ee4d40c3d900dc1f91d71febc93